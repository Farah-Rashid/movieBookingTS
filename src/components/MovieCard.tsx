import React from "react";
import { useNavigate } from "react-router";
import {Card, CardDiv }from "../styles/MovieCard.styled"


const imgUrl = "https://image.tmdb.org/t/p/w500";
const placeholderImg = "http://via.placeholder.com/1080x1580";

type Props = {
  movies: movie[];
};
type movie = {
  id:number;
  title:string;
  poster_path:string;
};

const MovieCard = (props: Props) => {
    const navigate=useNavigate();
  return (
    <CardDiv>
      {props.movies.map((item: movie) => {
        return (
          <Card key={item.id}>
            <h3>{item.title}</h3>
            {item.poster_path ? <img src={imgUrl + item.poster_path} alt={item.title} />: <img src={placeholderImg} alt="fakeImg"/>}
            <button onClick={()=>navigate("bookingseat",{state:{movie:imgUrl + item.poster_path,movieId:item.id}}) }>Book Now</button>
          </Card>
        );
      })}
    </CardDiv>
  );
};

export default MovieCard;
