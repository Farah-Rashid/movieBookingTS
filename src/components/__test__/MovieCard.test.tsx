import MovieCard from '../MovieCard'
import { render, screen, fireEvent } from "@testing-library/react"
import { BrowserRouter } from 'react-router-dom'

const data = [
  {
    id: 507086,
    title: "Jurassic World Dominion",
    poster_path: "/kAVRgw7GgK1CfYEJq8ME6EvRIgU.jpg",
  },
];

describe("testing movie cards", () => {
  test('should take a snapshot', () => {
    const { container } = render(<BrowserRouter><MovieCard movies={data} /></BrowserRouter>)
    expect(container).toMatchSnapshot()
    const Nav = screen.getByRole<HTMLAnchorElement>("link");
    expect(Nav).toHaveAttribute("href", "/book/507086/Jurassic World Dominion")
  })

  test("testing close button", () => {
    render(<BrowserRouter><MovieCard movies={data} /></BrowserRouter>);
    const primaryButton = screen.getByRole('button', { name: /Book now/i })
    fireEvent.click(primaryButton);
  })
})