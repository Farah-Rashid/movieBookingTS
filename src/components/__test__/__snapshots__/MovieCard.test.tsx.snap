// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing movie cards should take a snapshot 1`] = `
<div>
  <article
    class="sc-dkzDqf bjFjAH"
  >
    <article
      class="sc-gsnTZi jzTshG"
    >
      <h3>
        Jurassic World Dominion
      </h3>
      <img
        alt="Jurassic World Dominion"
        src="https://image.tmdb.org/t/p/w500//kAVRgw7GgK1CfYEJq8ME6EvRIgU.jpg"
      />
      <a
        class=""
        href="/book/507086/Jurassic World Dominion"
      >
        <button
          class="sc-bczRLJ hhvoiM"
          data-testid="buttonItem"
        >
          Book now
        </button>
      </a>
    </article>
  </article>
</div>
`;
